src/coreclr/jit/flowgraph.cpp

```diff
@@ -1683,12 +1683,9 @@ void Compiler::fgAddSyncMethodEnterExit()
 
     // Create a block for the start of the try region, where the monitor enter call
     // will go.
-
-    assert(fgFirstBB->bbFallsThrough());
-
-    BasicBlock* tryBegBB  = fgNewBBafter(BBJ_NONE, fgFirstBB, false);
-    BasicBlock* tryNextBB = tryBegBB->bbNext;
-    BasicBlock* tryLastBB = fgLastBB;
+    BasicBlock* const tryBegBB  = fgSplitBlockAtEnd(fgFirstBB);
```

Comment:
Why was the switch from `fgNewBBafter` to `fgSplitBlockAtEnd` required?

Reply:
Because in OSR methods `fgFirstBB` may not be `BBJ_NONE`.

Reply:
Even after calling `fgEnsureFirstBBisScratch()`?

Reply:
Yes, because `fgEnsureFirstBBisScratch` will just return if the first bb is already scratch.

For OSR we've already called this and messed with the resulting flow.  See `fgFixEntryFlowForOSR`.